crates/hyperqueue/src/server/client/autoalloc.rs

```diff
@@ -238,6 +242,206 @@ fn create_queue(
     }
 }
 
+struct QueueLimit {
+    walltime: Option<u32>,
+    backlog: Option<u32>,
+    workers: Option<u32>,
+}
+
+// Returns value of given arg, in formats: keyVALUE, key=VALUE, [key, VALUE]
+fn get_arg_by_key(key: String, args: &[String]) -> Option<String> {
+    let index_opt = args.iter().position(|r| r.contains(&key));
+
+    if let Some(index) = index_opt {
+        let mut value = args[index].clone();
+
+        value = value.replace(&key, "");
+        value = value.replace('=', "");
+
+        return if !value.is_empty() {
+            Some(value)
+        } else {
+            args.get(index + 1).cloned()
+        };
+    }
+    None
+}
+
+// Returns value of given path in json, can include nests
+fn get_value_by_key_json(
+    json: &serde_json::Value,
+    path: String,
+) -> anyhow::Result<&serde_json::Value> {
+    let json_keys = path.split('/');
+    let mut value = json;
+    for json_key in json_keys {
+        value = value
+            .get(json_key)
+            .context(format!("key {} not found", json_key))?;
+    }
+    Ok(value)
+}
+
+fn get_pbs_queue_limit(args: &[String]) -> anyhow::Result<Option<QueueLimit>> {
+    if let Some(q_type) = get_arg_by_key(String::from("-q"), args) {
+        let queue_type = q_type.replace("-q", "");
+        let queue_info = Command::new("qstat")
+            .arg("-Q")
+            .arg(&queue_type)
+            .arg("-f")
+            .arg("-F")
+            .arg("json")
+            .output()?;
+
+        let output = String::from_utf8_lossy(&queue_info.stdout).into_owned();
+        let json: serde_json::Value =
+            serde_json::from_str(&*output).expect("JSON was not well-formatted");
+        let queue = json
+            .get("Queue")
+            .and_then(|x| x.get(queue_type))
+            .context("can't find queue info")?;
+
+        let (walltime, backlog, workers) = match q_type.as_str() {
```

Comment:
We cannot depend on the specific queue names. Instead just try to find `resources_available/nodes` first and if it's missing, try `resources_max/nodect`.